test/test_script.py

```diff
@@ -1168,6 +1171,37 @@ def test_cmd_datetime(self):
             areg = r"^Wrong number of arguments for \$datetime: Expected between 0 and 1, "
             with self.assertRaisesRegex(ScriptError, areg):
                 self.parser.eval("$datetime(abc,def)")
+            # The following is disabled because it only raises the exception for Python < 3.7 on Ubuntu
+            # # Tests with invalid format code (ends with % failes on all platforms)
+            # areg = r"^\$datetime:\d+:\d+: Unsupported format code"
+            # with self.assertRaisesRegex(ScriptRuntimeError, areg):
+            #     self.parser.eval(r"$datetime(\%)")
+        except (AssertionError, ValueError, IndexError) as err:
+            raise err
+        finally:
+            # Restore original datetime object
+            datetime.datetime = original_datetime
+
+    @skipUnless(IS_WIN, "windows test")
```

Comment:
Instead of skipping on Windows, we should rather skip if underlying datetime format is not supported on the platform we test on.

Basically: execute `datetime.strftime(whatever)` and if it raises an exception, expect `$datetime(whatever)` raises `ScriptRuntimeError` as well.

Reply:
Similar to your other test suggestion above.  I like it.